internal/api/v1/costsheet.go

```diff
@@ -0,0 +1,354 @@
+package v1
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/flexprice/flexprice/internal/api/dto"
+	domainCostsheet "github.com/flexprice/flexprice/internal/domain/costsheet"
+	ierr "github.com/flexprice/flexprice/internal/errors"
+	"github.com/flexprice/flexprice/internal/logger"
+	"github.com/flexprice/flexprice/internal/service"
+	"github.com/flexprice/flexprice/internal/types"
+	"github.com/gin-gonic/gin"
+)
+
```

Comment:
For this file a general overall refactoring is required:
Move the business logic etc in the service methods itself